include/sound/sof/ext_manifest4.h

```diff
@@ -0,0 +1,118 @@
+/* SPDX-License-Identifier: ((GPL-2.0 WITH Linux-syscall-note) OR BSD-3-Clause) */
+/*
+ * This file is provided under a dual BSD/GPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2021 Intel Corporation. All rights reserved.
+ */
+
+/*
+ * Extended manifest is a place to store metadata about firmware, known during
+ * compilation time - for example firmware version or used compiler.
+ * Given information are read on host side before firmware startup.
+ * This part of output binary is not signed.
+ */
+
+#ifndef __SOF_FIRMWARE_EXT_MANIFEST4_H__
+#define __SOF_FIRMWARE_EXT_MANIFEST4_H__
+
+#include <linux/uuid.h>
+
+/* In ASCII  $AE1 */
+#define SOF_EXT_MAN4_MAGIC_NUMBER    0x31454124
+
+#define MAX_MODULE_NAME_LEN     8
+#define MAX_FW_BINARY_NAME      8
+#define DEFAULT_HASH_SHA256_LEN 32
+#define SOF_MAN4_FW_HDR_OFFSET	0x2000
+
+/*********************************************************************
+ *     extended manifest
+ *-------------------
+ *     css_manifest hdr
+ *-------------------
+ *     offset reserved for future
+ *-------------------
+ *     fw_hdr
+ *-------------------
+ *     module_entry[0]
+ *-------------------
+ *     module_entry[1]
+ *-------------------
+ *     ...
+ *-------------------
+ *     module_entry[n]
+ *-------------------
+ *     module_config[0]
+ *-------------------
+ *     module_config[1]
+ *-------------------
+ *     ...
+ *-------------------
+ *     module_config[m]
+ *-------------------
+ *     FW content
+ *-------------------
+ *********************************************************************/
+
+struct sof_ext_manifest4_hdr {
+	u32 id;
+	u32 len; /* length of extension manifest */
+	u16 version_major;
+	u16 version_minor;
```

Comment:
possible endianness issue?

Reply:
> possible endianness issue?

Yes, but do we need to support big endian? We will test `hdr->id == SOF_EXT_MAN4_MAGIC_NUMBER` in `snd_sof_fw_ext_man_parse` and it will be false if the firmware binary is built by the different type of endianness.
In other words, we will return error if the firmware binary is built by the different type of endianness.